spices/SPICE-0005-scheme-agnostic-projects.adoc

```diff
@@ -0,0 +1,185 @@
+= Scheme-agnostic Projects
+
+* Proposal: link:./SPICE-0005-scheme-agnostic-projects.adoc[SPICE-0005]
+* Author: link:https://github.com/bioball[Dan Chao]
+* Status: TBD
+* Implemented in: TBD
+* Category: Language
+
+== Introduction
+
+This SPICE proposes a way to allow projects to be usable within modules that are not files.
+
+== Motivation
+
+A _project_ is a mechanism that allows users to manage dependencies, common evaluator settings, and create packages.
+
+Currently, a project is also restricted to the local filesystem.
+For example, it is not possible to tell Pkl that the project directory is `alternativescheme:/path/to/my/project`.
+
+This limitation is a pain-point in some circumstances.
+
+* When Pkl is embedded as a library in Java, it is common to embed Pkl modules as Java resources, and load them via link:https://pkl-lang.org/main/current/language-reference/index.html#module-path-uri[modulepath] URIs.
+* When Pkl is embedded in Go, it is common to embed Pkl files using an embedded filesystem `go:embed` directive, and passed to Pkl using the `pkl.WithFs` option.
+
+Because these URIs are not files, it is not possible to use any project features in either of these two scenarios.
+
+== Proposed Solution
+
+There are to main changes to be made:
+
+1. Instead of a "project directory", a project is identified by a base URI.
+2. Relative paths starting with `@` are _always_ treated as link:https://pkl-lang.org/main/current/language-reference/index.html#dependency-notation-uris[dependency notation], in all modules.
+
+== Detailed design
```

Comment:
I'm missing a table/list/etc detailing for a project defined using module key `X` what are the allowed module keys inside the dependencies.
I think right now if you import `@foo/bar.pkl` then `bar.pkl` can only import other projects (using `@` or `package://`) or relative imports inside the project. If I now have a project defined with `modulepath` can I have `modulepath:` imports inside its dependencies?

Reply:
Summary from offline discussion: Projects/packages don't _remove_ things that can be imported, they can only add to what can be imported. Even normal projects whose project dir is `file:` based can have an `import modulepath:/` within its sources, and it still works today.